Update LSP6 relay call guide (#909)
lukso-network/docs: docs/learn/expert-guides/key-manager/execute-relay-transactions.md

## Setup

First, the transaction to be executed by a third party has to be prepared. This logic can be implemented:

- **client-side** and sent to a third-party application
- **service-side**, if the Transaction Relay Service is set up on the Universal Profile

You will need the following dependencies installed:

- a blockchain provider library like [`ethers`](https://github.com/ethers-io/ethers.js/) or [`web3`](https://github.com/web3/web3.js)
- the [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/) package to get the artifacts of the [LSP contracts](../../../standards/introduction.md)
- the [`@lukso/eip191-signer.js`](https://github.com/lukso-network/tools-eip191-signer) library to sign the relay transaction

<Tabs groupId="provider-lib">

  <TabItem value="ethers" label="ethers">

```shell
npm install ethers @lukso/lsp-smart-contracts @lukso/eip191-signer.js
```

  </TabItem>

  <TabItem value="web3" label="web3">

```shell
npm install web3 @lukso/lsp-smart-contracts @lukso/eip191-signer.js
```

  </TabItem>

</Tabs>

:::info Execution Rights

> Even when relaying the transaction using a third-party, you will need execute permission is necessary. Otherwise, unauthorized parties would be able to send transactions to your relayer address.

Reviewer: Suggest adding link to the permission in the LSP6 standard page. And remove the 2nd sentence which contradicts the first.

```suggestion
To execute relay transactions using a third-party add, you will need the `EXECUTE_RELAY_CALL` permission.
```

- **client-side** and sent to a third-party application
- **service-side**, if the Transaction Relay Service is set up on the Universal Profile

You will need the following dependencies installed:

- a blockchain provider library like [`ethers`](https://github.com/ethers-io/ethers.js/) or [`web3`](https://github.com/web3/web3.js)
- the [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/) package to get the artifacts of the [LSP contracts](../../../standards/introduction.md)
- the [`@lukso/eip191-signer.js`](https://github.com/lukso-network/tools-eip191-signer) library to sign the relay transaction

<Tabs groupId="provider-lib">

  <TabItem value="ethers" label="ethers">

```shell
npm install ethers @lukso/lsp-smart-contracts @lukso/eip191-signer.js
```

  </TabItem>

  <TabItem value="web3" label="web3">

```shell
npm install web3 @lukso/lsp-smart-contracts @lukso/eip191-signer.js
```

  </TabItem>

</Tabs>

:::info Execution Rights

To execute relay transactions using a third-party add, you will need the [`EXECUTE_RELAY_CALL`](https://docs.lukso.tech/standards/universal-profile/lsp6-key-manager/#permissions) permission.

:::

> After installation, we can import the contract artifacts and set up relevant constants. To encode the transaction, you need the address of the Universal Profile smart contract and the private key of a controller key with sufficient [LSP6 Permissions](../../../standards/universal-profile/lsp6-key-manager.md#permissions) to **execute** the transaction. The controller, addresses, and artifacts will then be used to set up the contract instances for the [Universal Profile](../../../standards/universal-profile/lsp0-erc725account.md) and [Key Manager](../../../standards/universal-profile/lsp6-key-manager.md) you will interact with.

Reviewer: Suggest rewording back to how it was, and maybe add 1 extra sentence:

```suggestion
To encode a transaction, we need the address of the Universal Profile smart contract and the private key of a controller key with sufficient [LSP6 permissions](../../../standards/universal-profile/lsp6-key-manager.md#permissions) to execute the transaction.
First, create an instance of the Universal Profile contract and its Key Manager.
```

```typescript
import Web3 from 'web3';

import UniversalProfileContract from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
import KeyManagerContract from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json';
import { EIP191Signer } from '@lukso/eip191-signer.js';

// This version is relative to the LSP25 standard, defined as 25.
import { LSP25_VERSION } from '@lukso/lsp-smart-contracts/constants';

const web3 = new Web3('https://rpc.testnet.lukso.network');
const universalProfileAddress = '0x...';
const recipientAddress = '0x...';

// Setup the Universal Profile controller account
const controllerPrivateKey = '0x...';
const controllerAccount = web3.eth.accounts.wallet.add(controllerPrivateKey);

// Setup the contract instance of the Universal Profile
const universalProfile = new web3.eth.Contract(
  UniversalProfileContract.abi,
  universalProfileAddress,
);

// Call the Universal Profile contract to get the Key Manager
const keyManagerAddress = await universalProfile.methods.owner().call();

// Setup the contract instance of the Key Manager
const keyManager = new web3.eth.Contract(
  KeyManagerContract.abi,
  keyManagerAddress,
);
```

  </TabItem>

</Tabs>

> :::caution Caution when using a Private Key

Reviewer: This is a good callout 👌 . Would even put that in danger.

```suggestion
:::danger Caution when using your controller's Private Key
```

Never share your private controller key or upload it to public repositories. Anyone who possesses it can access your funds and assets and gain control over your Universal Profile in case the controller has administrative rights!

:::

## Prepare the Relay Call

After setting up both the [Universal Profile](../../../standards/universal-profile/introduction.md) and [Key Manager](../../../standards/universal-profile/lsp6-key-manager.md) of the signing person, we can continue to prepare all the relay call parameters. These are crucial for ensuring secure and authenticated transactions. Here is what you will need:

> - **`nonce` of the controller**: This value can be retrieved by calling the [`getNonce`](../../../contracts/contracts/LSP6KeyManager/LSP6KeyManager.md#getnonce) function on the [Key Manager](../../../standards/universal-profile/lsp6-key-manager.md) associated with the profile. The `nonce` of an EOA is incremented with each transaction, ensuring their uniqueness.

Reviewer: The nonce can be from an EOA or a contract. Suggest removing this

```suggestion
- **`nonce` of the controller**: can be retrieved via the [`getNonce`](../../../contracts/contracts/LSP6KeyManager/LSP6KeyManager.md#getnonce) function on the [Key Manager](../../../standards/universal-profile/lsp6-key-manager.md) associated with the Universal Profile.
```

    // Encoded value: `0x0000000000000000000000000000000000000000000000000000000000000005`
    nonce,          
                      
    // e.g: valid until 1st January 2025 at midnight (GMT).
    // Timestamp = 1735689600
    // Encoded value: `0x0000000000000000000000000000000000000000000000000000000067748580`
    validityTimestamps,      

    // e.g: not funding the contract with any LYX (0)
    // Encoded value: `0x0000000000000000000000000000000000000000000000000000000000000000`
    msgValue,    
           
    // e.g: send 3 LYX to address 0xcafecafecafecafecafecafecafecafecafecafe
    // by calling execute(uint256,address,uint256,bytes)
    // Encoded value: `0x44c028fe00000000000000000000000000000000000000000000000000000000        
    //                 00000000000000000000000000000000cafecafecafecafecafecafecafecafeca 
    //                 fecafecafecafe00000000000000000000000000000000000000000000000029a2 
    //                 241af62c0000000000000000000000000000000000000000000000000000000000
    //                 000000008000000000000000000000000000000000000000000000000000000000
    //                 00000000`
    abiPayload,         
  ],
);

> // Instanciate EIP191 Signer

Reviewer: ```suggestion
// Instantiate EIP191 Signer
```

const eip191Signer = new EIP191Signer();

> /**

Reviewer: Suggest removing this code comment block for brevity. The code and function parameters are self explanatory themselves.
